src/nbl/video/IDescriptorPool.cpp

```diff
@@ -4,64 +4,166 @@
 namespace nbl::video
 {
 
-bool IDescriptorPool::createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
+uint32_t IDescriptorPool::createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
 {
-    core::vector<SDescriptorOffsets> descriptorOffsets(count);
+    core::vector<SDescriptorOffsets> descriptorOffsets;
+    descriptorOffsets.reserve(count);
+
+    system::ILogger* logger = nullptr;
+    {
+        auto debugCallback = getOriginDevice()->getPhysicalDevice()->getDebugCallback();
+        if (debugCallback)
+            logger = debugCallback->getLogger();
+    }
 
     for (uint32_t i = 0u; i < count; ++i)
     {
         if (!isCompatibleDevicewise(layouts[i]))
-            return false;
+        {
+            if (logger)
+                logger->log("Device-Incompatible descriptor set layout found at index %u. Sets for the layouts following this index will not be created.", system::ILogger::ELL_WARNING, i);
+
+            break;
+        }
         
-        descriptorOffsets[i] = allocateDescriptorOffsets(layouts[i]);
+        if (!allocateDescriptorOffsets(descriptorOffsets.emplace_back(), layouts[i]))
+        {
+            if (logger)
+                logger->log("Failed to allocate descriptor offsets in the pool's storage for descriptor set layout at index %u. Sets for the layouts following this index will not be created.", system::ILogger::ELL_WARNING, i);
+
+            descriptorOffsets.pop_back();
+            break;
+        }
+    }
+
+    const auto successCount = descriptorOffsets.size();
+    assert(count >= successCount);
+    std::fill_n(output + successCount, count - successCount, nullptr);
+
+    const uint32_t allocatedDSOffset = m_descriptorSetAllocator.alloc_addr(successCount, 1u);
```

Comment:
no this is not how a Pool Address allocator works, you cannot allocate multiple contiguous areas, you literally need to do multiple `alloc_addr` calls (1 per set)

and probably hoist this to line 28 or something

Reply:
actually would be best to fold the DS offset allocation into `allocateDescriptorOffsets` and `SDescriptorOffsets`

Reply:
Resolved.